source/Ship.cpp

```diff
@@ -3446,3 +3459,17 @@ void Ship::CreateSparks(vector<Visual> &visuals, const Effect *effect, double am
 			visuals.emplace_back(*effect, angle.Rotate(point) + position, velocity, angle);
 	}
 }
+
+// Determine if the ship has an outfit with the specified type and optionally subtype
+bool Ship::HasOutfit(const string &type, const string &subtype) const
```

Comment:
New functions should be placed in the same spot as in the header files. i.e. don't put this at the bottom of the file, but it below AddOutfit.

Reply:
I will make the correction and post a commit with the update.